libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -846,6 +846,18 @@ - (void)setupWithCredentials:(SFOAuthCredentials*) credentials {
     self.idCoordinator.delegate = self;
 }
 
+- (void)restartAuthentication {
+    @synchronized (self.authBlockList) {
+        if (!self.authenticating) {
+            [self log:SFLogLevelWarning format:@"%@: Authentication manager is not currently authenticating.  No action taken.", NSStringFromSelector(_cmd)];
+            return;
+        }
+        [self log:SFLogLevelInfo format:@"%@: Restarting in-progress authentication process.", NSStringFromSelector(_cmd)];
+        [self.coordinator stopAuthentication];
+        [self loginWithCredentials:[self createOAuthCredentials]];
```

Comment:
The property is already effectively stored, internally.  The issue is that, for this use case, you have to change it prior to restarting authentication, because the credentials object owns the login host at the time that authentication actually takes place.

So my approach for changing it was to create new credentials, which reconfigures with the (updated) login host, since in the use case we care about, the previous credentials are temporary anyway.

If we would want to re-use the existing credentials, we'd need another way to tell configure the object to use the new login host.  But in that scenario it doesn't really make sense, because existing credentials should be permanently tied to whatever login host they had in the first place.

Which is basically what I was getting at in my second point: this process should be reasonably hard-wired between a login host change from the user interface and an associated auth restart—there's no reason to initiate that auth restart in any other scenario I can think of.  Which is what led me to consider whether we should just use a callback approach (delegate, notification), which would satisfy that specificity of process.